onnxscript/ir/_external_data.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+"""Pass to save tensor data as external tensors."""
+
+import os
+
+from onnxscript.ir import _core
+
+
+class ExternalDataInfo:
+    def __init__(self, name, offset, length):
+        self.name = name
+        self.offset = offset
+        self.length = length
+
+
+def save_external_data(
+    initializers,
+    file_path,
+    allocation_granularity: int = 65536,  # 64KB
+):
+    # Store external data such as name, offset and size
+    external_data_info = dict()
+
+    # Create file if it doesn't exist
+    if not os.path.isfile(file_path):
+        with open(file_path, "ab"):
+            pass
+
+    with open(file_path, "r+b") as data_file:
+        current_offset = 0
+        for i_name, i_value in initializers.items():
+            tensor_val = i_value
+            if isinstance(tensor_val, _core.Value):
+                tensor_val = i_value.const_value()  # pylint: disable=protected-access
+            assert isinstance(tensor_val, _core.Tensor)
```

Comment:
It can be any object conforming to ir.TensorProtocol, so a type checking is not needed.